openslides_gui/gui.py

```diff
@@ -0,0 +1,763 @@
+
+    def save_gui_settings(self):
+        if self.last_backup is None:
+            last_backup = None
+        else:
+            last_backup = self.last_backup.strftime("%Y-%m-%d %H:%M:%S")
+        settings = {
+            "database_backup": {
+                "enabled": self.backupdb_enabled,
+                "destination": self.backupdb_destination,
+                "internal": self.backupdb_interval,
+                "interval_unit": self.backupdb_interval_unit,
+                "last_backup": last_backup
+            },
+            "server_settings": {
+                "host": self.host,
+                "port": self.port,
+            },
+        }
+
+        dp = os.path.dirname(self.gui_settings_path)
+        if not os.path.exists(dp):
+            os.makedirs(dp)
+        with open(self.gui_settings_path, "w", encoding="utf-8") as f:
+            json.dump(settings, f, ensure_ascii=False, indent=4)
+
+    def apply_backup_settings(self):
+        if self.backupdb_enabled and self.server_running:
+            now = datetime.datetime.utcnow()
+            delta = datetime.timedelta(seconds=self.backup_interval_seconds)
+            ref = self.last_backup
+            if ref is None:
+                ref = now
+            ref += delta
+
+            d = ref - now
+            seconds = d.days * 86400 + d.seconds
+            if seconds < 1:
+                seconds = 30  # avoid backup immediatly after start
+            self.backup_timer.Start(seconds * 1000, True)
+        else:
+            self.backup_timer.Stop()
+
+    def do_backup(self):
+        cmd = [
+            sys.executable, "-u", "-m", "openslides", "backupdb",
+            self.backupdb_destination,
+        ]
+        p = subprocess.Popen(
+            cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT)
+        p.stdin.close()
+        output = p.stdout.read().strip()
+        exitcode = p.wait()
+        if output:
+            self.cmd_run_ctrl.append_message(output)
+
+        time = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        if exitcode == 0:
+            self.cmd_run_ctrl.append_message(
+                _("{0}: Database backup successful.").format(time))
+        else:
+            self.cmd_run_ctrl.append_message(
+                _("{0}: Database backup failed!").format(time))
+
+        self.last_backup = datetime.datetime.utcnow()
+
+    def on_syncdb_clicked(self, evt):
+        self.cmd_run_ctrl.append_message(_("Syncing database..."))
+        self.cmd_run_ctrl.run_command("syncdb")
+
+    def on_reset_admin_clicked(self, evt):
+        self.cmd_run_ctrl.append_message(_("Resetting admin user..."))
+        self.cmd_run_ctrl.run_command("createsuperuser")
+
+    def on_about_clicked(self, evt):
+        info = wx.adv.AboutDialogInfo()
+        info.SetName("OpenSlides")
+        # TODO: get_version was removed - replace with new method
+        # info.SetVersion(openslides.get_version())
```

Comment:
Please use for version_str:
```python
import openslides
print(openslides.__version__)
```
see https://github.com/OpenSlides/OpenSlides/issues/1492#issuecomment-99356146